ocaml/xapi/xapi_event.ml

```diff
@@ -604,6 +604,16 @@ let inject ~__context ~_class ~_ref =
 
 (* Internal interface ****************************************************)
 
+let create_call_task ~__context loc =
+  debug "Creating wakeup for %s" loc;
+  let task_id = (TaskHelper.XapiEvent ("TaskForEvents: " ^ loc)) in
```

Comment:
This doesn't give you a task in the DB. You could use `Xapi_task.create`.

Reply:
By the way, `TaskHelper` is weirdly entangled with the `Context` module... I've got some patches to clean it up a little, but that's for later.

Reply:
Using Xapi_task from here creates a cyclic dependency. It should still be possible, but might need a bit of a shuffle

Reply:
Or just do the same as that function does, but inline, e.g.:
```
let task_id =
  Context.make ~task_description:"something useful" ~task_in_database:true "suitable name"
  |> Context.get_task_id
in
```